lib/uri/common.rb

```diff
@@ -46,7 +46,7 @@ def self.parser=(parser = RFC3986_PARSER)
   self.parser = RFC3986_PARSER
 
   def self.const_missing(const)
-    if value = RFC2396_PARSER.regexp[const]
+    if value == RFC2396_PARSER.regexp[const]
```

Comment:
This change is wrong.  It is supposed to set a local variable, used later, not check for equality.